File: mobile/lib/features/cash_wallets/models/cash_wallet.dart
class CashWallet {
  const CashWallet({
    required this.id,
    required this.name,
    required this.balance,
    required this.currency,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String name;
  final double balance;
  final String currency;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory CashWallet.fromJson(Map<String, dynamic> json) {
    return CashWallet(
      id: json['id'] as String,
      name: json['name'] as String,
      balance: (json['balance'] as num).toDouble(),
      currency: json['currency'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }
}

File: api/app/schemas/cash_wallet.py
from datetime import datetime

from pydantic import BaseModel, Field


class CashWalletCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    balance: float = Field(default=0, ge=0)
    currency: str = Field(default="USD", min_length=3, max_length=3)


class CashWalletUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=80)
    balance: float | None = Field(default=None, ge=0)
    currency: str | None = Field(default=None, min_length=3, max_length=3)


class CashWalletResponse(BaseModel):
    id: str
    name: str
    balance: float
    currency: str
    created_at: datetime
    updated_at: datetime
